samples/show-mobile-map-package-expiration-date/src/main/java/com/esri/arcgismaps/sample/showmobilemappackageexpirationdate/components/ShowMobileMapPackageExpirationDateViewModel.kt

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showmobilemappackageexpirationdate.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.MobileMapPackage
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import com.esri.arcgismaps.sample.showmobilemappackageexpirationdate.R
+import java.io.File
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import java.util.Locale
+import kotlinx.coroutines.launch
+
+class ShowMobileMapPackageExpirationDateViewModel(application: Application) : AndroidViewModel(application) {
+    // The map displayed on the MapView. Updated after loading the mobile map package
+    var arcGISMap: ArcGISMap by mutableStateOf(ArcGISMap())
+        private set
+
+    // Message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Expiration UI states exposed to the screen
+    var isExpired by mutableStateOf(false)
+        private set
+
+    var expirationMessage by mutableStateOf("")
+        private set
+
+    var expirationDateText by mutableStateOf("N/A")
+        private set
+
+    // Build the provision path where offline resources are stored for the sample app
+    private val provisionPath: String by lazy {
+        val basePath = application.getExternalFilesDir(null)?.path ?: ""
+        val appFolderName = getApplication<Application>().getString(
+            R.string.show_mobile_map_package_expiration_date_app_name
+        )
```

Comment:
Use the ViewModel provided `app` to build path: 

```suggestion
        val basePath = app.getExternalFilesDir(null)?.path.toString()
        val appFolderName = app.getString(R.string.show_mobile_map_package_expiration_date_app_name)
```